README.md

```diff
@@ -14,17 +15,27 @@ gem 'annealing'
 
 And then execute:
 
-    $ bundle install
+```shell
+bundle install
+```
 
 Or install it yourself as:
 
-    $ gem install annealing
+```shell
+gem install annealing
+```
 
 ## Usage
 
-To use this gem, you must provide a lambda function to calculate the total energy of an object, and a way to change the state of the object to a neighboring state.
+Simulated annealing algorithms work by comparing multiple permutations of a given object and measuring their relative efficiencies based on any number of competing factors. If you aren't already familiar with the concept of simulated annealing, we recommend watching [The Most Metal Algorithm in Computer Science](https://www.youtube.com/watch?v=I_0GBWCKft8) from [SciShow](https://www.youtube.com/c/SciShow) as it will help you understand some of the concepts and terms used below.
```

Comment:
Am I remembering correctly that you used to have a parent repo to this that included an elixer implementation as well as a link to this video? I can't find that anymore, but I found that video incredibly helpful for understanding simulated annealing when I was starting to dig into Ruby implementations. I thought it would be useful to include here to help introduce all the terms and why certain arguments and classes are named the way they are.